jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -97,30 +105,51 @@ def __init__(self, *args, **kwargs):
         self.session.hub.subscribe(self, AddDataToViewerMessage,
                                    handler=self._find_and_convert_contour_units)
 
+        self.has_spectral = self.config in ('specviz', 'cubeviz', 'specviz2d', 'mosviz')
         self.spectral_unit = UnitSelectPluginComponent(self,
                                                        items='spectral_unit_items',
                                                        selected='spectral_unit_selected')
 
-        self.spectral_y_type = SelectPluginComponent(self,
-                                                     items='spectral_y_type_items',
-                                                     selected='spectral_y_type_selected',
-                                                     manual_options=['Surface Brightness', 'Flux'])
-
+        self.has_flux = self.config in ('specviz', 'cubeviz', 'specviz2d', 'mosviz')
         self.flux_unit = UnitSelectPluginComponent(self,
                                                    items='flux_unit_items',
                                                    selected='flux_unit_selected')
 
+        self.has_angle = self.config in ('cubeviz', 'specviz', 'mosviz')
         self.angle_unit = UnitSelectPluginComponent(self,
                                                     items='angle_unit_items',
                                                     selected='angle_unit_selected')
 
+        self.has_sb = self.has_angle or self.config in ('imviz',)
```

Comment:
```suggestion
        self._has_sb = self.has_angle or self.config in ('imviz',)
```

As per "convention," should this be "hidden" then if read only?

Reply:
Traitlets cannot have leading underscores, and this is not exposed to the user